### PR TITLE
feat: WASM build on NPM

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Summary:**  
Added wasm build that can be used in browser

### Node
Steps to publish:`cd node && npm run publish`
Steps to run example: `cd node/example && npm run serve`

### Browser
Steps to publish:`cd browser && npm run publish`
Steps to run example: `cd browser` and you can simply use `index.html`

refer: https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_Wasm

**Issue Reference(s):**  
Fixes: #1038 
/claim 1038

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcall/tree/main/docs

### CODE REVIEW

File: node/src/cache.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    async fn get<'a>(&'a self, key: &'a Self::Key) -> anyhow::Result<Self::Value> {
+        Ok(self
+            .cache
+            .get(key.clone())
+            .ok_or(anyhow!("No such key found"))?
+            .value()
+            .clone())
+    }
+}
```
Comment (tusharmath) [inline node/src/cache.rs:39]: can use `src/cache.rs`

File: browser/index.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "\n" +
+        "type Post {\n" +
+        "  id: Int!\n" +
+        "  userId: Int!\n" +
+        "  title: String!\n" +
+        "  body: String!\n" +
+        '  user: User @http(path: "/users/{{value.userId}}")\n' +
+        "}\n"
+      const source = ".graphql"
+      const executor = new GraphQLExecutor(schema, source)
```
Comment (tusharmath) [inline browser/index.html:45]:
```suggestion
      const executor = new GraphQLExecutor({graphql: schema})
```

For other extensions

```js
const executor = new GraphQLExecutor({json: schema})
const executor = new GraphQLExecutor({yml: schema})
```

File: node/src/env.rs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+pub struct WasmEnv {}
+impl WasmEnv {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+impl EnvIO for WasmEnv {
+    fn get(&self, key: &str) -> Option<String> {
+        unimplemented!("Not needed for npm pkg")
```
Comment (tusharmath) [inline node/src/env.rs:11]: Not ok. There should be a way to support this in the browser.

File: browser/index.html (hunk 2 above)
Comment (tusharmath) [reply browser/index.html:45]: Pass env here.

```js
new GraphQLExecutor({env: {foo: "bar"})
```

File: node/src/env.rs (hunk 3 above)
Comment (ologbonowiwi) [reply node/src/env.rs:11]: On node we do have process.env, ideally we should read from it

**Pull request closed without merging**